lib/containers/docker.pm

```diff
@@ -32,4 +33,56 @@ sub configure_insecure_registries {
     record_info "setup $self->runtime", "deamon.json ready";
 }
 
+sub check_containers_firewall {
+    record_info "firewall", "Checking that firewall is enabled and properly configured";
+    my $container_name = 'sut_container';
+    my $docker_version = get_docker_version();
+    systemctl('is-active firewalld');
+    my $running = script_output qq(docker ps -q | wc -l);
+    validate_script_output('ip a s docker0', sub { /state DOWN/ }) if $running == 0;
+    # Docker zone is created for docker version >= 20.10 (Tumbleweed), but it
+    # is backported to docker 19 for SLE15-SP3 and for Leap 15.3
+    if (check_runtime_version($docker_version, ">=20.10") || is_sle('>=15-sp3') || is_leap(">=15.3")) {
+        assert_script_run "firewall-cmd --list-all --zone=docker";
+        validate_script_output "firewall-cmd --list-interfaces --zone=docker", sub { /docker0/ };
+        validate_script_output "firewall-cmd --get-active-zones", sub { /docker/ };
+    }
+    # Rules applied before DOCKER. Default is to listen to all tcp connections
+    # ex. output: "1           0        0 RETURN     all  --  *      *       0.0.0.0/0            0.0.0.0/0"
+    validate_script_output "iptables -L DOCKER-USER -nvx --line-numbers", sub { /1.+all.+0\.0\.0\.0\/0\s+0\.0\.0\.0\/0/ };
+
+    # Run container in the background
+    assert_script_run "docker run -id --rm --name $container_name -p 1234:1234 " . registry_url('alpine');
+    my $container_ip = container_ip($container_name, 'docker');
+
+    # Each running container should have added a new entry to the DOCKER zone.
+    # ex. output: "1           0        0 ACCEPT     tcp  --  !docker0 docker0  0.0.0.0/0            172.17.0.2           tcp dpt:1234"
+    validate_script_output "iptables -L DOCKER -nvx --line-numbers", sub { /1.+ACCEPT.+!docker0 docker0.+$container_ip\s+tcp dpt:1234/ };
+
+    # Kill the container running on background
+    assert_script_run "docker kill $container_name ";
```

Comment:
Killing a container which does not exists (you use the `--rm` parameter before) will result in a failure:

    $ docker run --rm -ti --name hale-bopp opensuse/tumbleweed
    418a496d900f:/ # exit
    $ docker kill hale-bopp
    Error response from daemon: Cannot kill container: hale-bopp: No such container: hale-bopp

So either add a check if the container exists or change to `script_run` instead.

Reply:
actually, it works if you use Alpine. I guess the key is in the ENTRYPOINT. In your example, it executes "bash", so it enters the container. If you do the same with Alpine, it keeps it running.

Reply:
Yes, the container runs in the background as the comment suggests but I will add a check for it.

Reply:
The difference is the `-it` vs. `-id`. The `-d` makes it run in the background. There shouldn't be a difference between alpine or TW there.

Reply:
I added `sleep 30d` to the `podman run` command so it's more visible that it is running in the background. Can you please re-check?

Reply:
Now it makes sense. Since this is a cleanup, I'd make it a `script_run` instead of an `assert_script_run`, however this is up to you.